src/status_im/accounts/access/core.cljs

```diff
@@ -0,0 +1,105 @@
+(ns status-im.accounts.access.core
+  (:require [clojure.string :as string]
+            [re-frame.core :as re-frame]
+            [status-im.accounts.create.core :as accounts.create]
+            [status-im.accounts.db :as db]
+            [status-im.i18n :as i18n]
+            [status-im.native-module.core :as status]
+            [status-im.ui.screens.navigation :as navigation]
+            [status-im.utils.ethereum.mnemonic :as mnemonic]
+            [status-im.utils.identicon :as identicon]
+            [status-im.utils.security :as security]
+            [status-im.utils.types :as types]
+            [status-im.utils.fx :as fx]))
+
+(defn check-phrase-errors [recovery-phrase]
+  (cond (string/blank? recovery-phrase) :required-field
+        (not (mnemonic/valid-phrase? recovery-phrase)) :recovery-phrase-invalid))
+
+(defn check-phrase-warnings [recovery-phrase]
+  (when (not (mnemonic/status-generated-phrase? recovery-phrase))
+    :recovery-phrase-unknown-words))
+
+(fx/defn set-phrase [{:keys [db]} masked-recovery-phrase]
+  (let [recovery-phrase (security/unmask masked-recovery-phrase)]
+    {:db (update db :accounts/access assoc :passphrase (string/lower-case recovery-phrase))}))
+
+(fx/defn validate-phrase [{:keys [db]}]
+  (let [recovery-phrase (get-in db [:accounts/access :passphrase])]
+    {:db (update db :accounts/access assoc
+                 :passphrase-error (check-phrase-errors recovery-phrase)
+                 :passphrase-warning (check-phrase-warnings recovery-phrase))}))
+
+(defn access-account! [masked-passphrase password]
+  (status/access-account
+   (mnemonic/sanitize-passphrase (security/unmask masked-passphrase))
+   password
+   (fn [result]
+     ;; here we deserialize result, dissoc mnemonic and serialize the result again
+     ;; because we want to have information about the result printed in logs, but
+     ;; don't want secure data to be printed
+     (let [data (-> (types/json->clj result)
+                    (dissoc :mnemonic)
+                    (types/clj->json))]
+       (re-frame/dispatch [:accounts.access.callback/access-account-success data password])))))
+
+(fx/defn validate-access-result [{:keys [db] :as cofx} {:keys [error pubkey address]} password]
+  (if (empty? error)
+    (let [account {:pubkey     pubkey
+                   :address    address
+                   :photo-path (identicon/identicon pubkey)
+                   :mnemonic   ""}]
+      (accounts.create/on-account-created cofx account password true))
+    {:db (assoc-in db [:accounts/access :error] (i18n/label :t/recover-password-invalid))}))
+
+(fx/defn on-account-accessed [cofx result password]
+  (let [data (types/json->clj result)]
+    (fx/merge cofx
+              (validate-access-result data password))))
```

Comment:
There is no need to use `fx/merge` with only one function, could be just `(validate-access-result cofx data password)`

Reply:
agree, will update in a couple of mins